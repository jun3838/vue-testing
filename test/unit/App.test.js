import {shallow} from 'vue-test-utils';
import {createRenderer} from 'vue-server-renderer';

import App from '@/App.vue';

describe('App.vue', () => {
  it('should render', () => {
    const wrapper = shallow(App);
    expect(wrapper.find('#app')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const render = createRenderer();
    const app = shallow(App);

    expect(app).toMatchSnapshot();
  });
});
