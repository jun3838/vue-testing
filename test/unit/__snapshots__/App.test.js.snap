// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "element": <div
    id="app"
  >
    <router-view />
  </div>,
  "isVueComponent": true,
  "options": Object {
    "attachedToDocument": false,
  },
  "update": [Function],
  "version": 2.5,
  "vm": VueComponent {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$el": <div
      id="app"
    >
      <router-view />
    </div>,
    "$emit": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_Ctor": Object {
        "3": [Function],
      },
      "_base": [Function],
      "_parentElm": null,
      "_refElm": null,
      "components": Object {},
      "directives": Object {},
      "filters": Object {},
      "inject": Object {},
      "name": "app",
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {},
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": true,
    "_isVue": true,
    "_renderProxy": VueComponent {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        id="app"
      >
        <router-view />
      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "3": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_refElm": null,
        "components": Object {},
        "directives": Object {},
        "filters": Object {},
        "inject": Object {},
        "name": "app",
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_renderProxy": [Circular],
      "_self": [Circular],
      "_uid": 1,
      "_vnode": VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": Array [
          VNode {
            "asyncFactory": undefined,
            "asyncMeta": undefined,
            "children": undefined,
            "componentInstance": undefined,
            "componentOptions": undefined,
            "context": [Circular],
            "data": undefined,
            "elm": <router-view />,
            "functionalContext": undefined,
            "functionalOptions": undefined,
            "functionalScopeId": undefined,
            "isAsyncPlaceholder": false,
            "isCloned": false,
            "isComment": false,
            "isOnce": false,
            "isRootInsert": false,
            "isStatic": false,
            "key": undefined,
            "ns": undefined,
            "parent": undefined,
            "raw": false,
            "tag": "router-view",
            "text": undefined,
          },
        ],
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": [Circular],
        "data": Object {
          "attrs": Object {
            "id": "app",
          },
        },
        "elm": <div
          id="app"
        >
          <router-view />
        </div>,
        "functionalContext": undefined,
        "functionalOptions": undefined,
        "functionalScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": true,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "div",
        "text": undefined,
      },
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
    },
    "_self": [Circular],
    "_uid": 1,
    "_vnode": VNode {
      "asyncFactory": undefined,
      "asyncMeta": undefined,
      "children": Array [
        VNode {
          "asyncFactory": undefined,
          "asyncMeta": undefined,
          "children": undefined,
          "componentInstance": undefined,
          "componentOptions": undefined,
          "context": [Circular],
          "data": undefined,
          "elm": <router-view />,
          "functionalContext": undefined,
          "functionalOptions": undefined,
          "functionalScopeId": undefined,
          "isAsyncPlaceholder": false,
          "isCloned": false,
          "isComment": false,
          "isOnce": false,
          "isRootInsert": false,
          "isStatic": false,
          "key": undefined,
          "ns": undefined,
          "parent": undefined,
          "raw": false,
          "tag": "router-view",
          "text": undefined,
        },
      ],
      "componentInstance": undefined,
      "componentOptions": undefined,
      "context": [Circular],
      "data": Object {
        "attrs": Object {
          "id": "app",
        },
      },
      "elm": <div
        id="app"
      >
        <router-view />
      </div>,
      "functionalContext": undefined,
      "functionalOptions": undefined,
      "functionalScopeId": undefined,
      "isAsyncPlaceholder": false,
      "isCloned": false,
      "isComment": false,
      "isOnce": false,
      "isRootInsert": true,
      "isStatic": false,
      "key": undefined,
      "ns": undefined,
      "parent": undefined,
      "raw": false,
      "tag": "div",
      "text": undefined,
    },
    "_watcher": Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {},
      "deps": Array [],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 2,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": false,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
    "_watchers": Array [
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    ],
  },
  "vnode": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": Array [
      VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": undefined,
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": VueComponent {
          "$attrs": Object {},
          "$children": Array [],
          "$createElement": [Function],
          "$el": <div
            id="app"
          >
            <router-view />
          </div>,
          "$emit": [Function],
          "$listeners": Object {},
          "$options": Object {
            "_Ctor": Object {
              "3": [Function],
            },
            "_base": [Function],
            "_parentElm": null,
            "_refElm": null,
            "components": Object {},
            "directives": Object {},
            "filters": Object {},
            "inject": Object {},
            "name": "app",
            "render": [Function],
            "staticRenderFns": Array [],
          },
          "$parent": undefined,
          "$refs": Object {},
          "$root": [Circular],
          "$scopedSlots": Object {},
          "$slots": Object {},
          "$vnode": undefined,
          "_c": [Function],
          "_data": Object {},
          "_directInactive": false,
          "_events": Object {},
          "_hasHookEvent": false,
          "_inactive": null,
          "_isBeingDestroyed": false,
          "_isDestroyed": false,
          "_isMounted": true,
          "_isVue": true,
          "_renderProxy": VueComponent {
            "$attrs": Object {},
            "$children": Array [],
            "$createElement": [Function],
            "$el": <div
              id="app"
            >
              <router-view />
            </div>,
            "$emit": [Function],
            "$listeners": Object {},
            "$options": Object {
              "_Ctor": Object {
                "3": [Function],
              },
              "_base": [Function],
              "_parentElm": null,
              "_refElm": null,
              "components": Object {},
              "directives": Object {},
              "filters": Object {},
              "inject": Object {},
              "name": "app",
              "render": [Function],
              "staticRenderFns": Array [],
            },
            "$parent": undefined,
            "$refs": Object {},
            "$root": [Circular],
            "$scopedSlots": Object {},
            "$slots": Object {},
            "$vnode": undefined,
            "_c": [Function],
            "_data": Object {},
            "_directInactive": false,
            "_events": Object {},
            "_hasHookEvent": false,
            "_inactive": null,
            "_isBeingDestroyed": false,
            "_isDestroyed": false,
            "_isMounted": true,
            "_isVue": true,
            "_renderProxy": [Circular],
            "_self": [Circular],
            "_uid": 1,
            "_vnode": [Circular],
            "_watcher": Watcher {
              "active": true,
              "cb": [Function],
              "deep": false,
              "depIds": Set {},
              "deps": Array [],
              "dirty": false,
              "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
              "getter": [Function],
              "id": 2,
              "lazy": false,
              "newDepIds": Set {},
              "newDeps": Array [],
              "sync": false,
              "user": false,
              "value": undefined,
              "vm": [Circular],
            },
            "_watchers": Array [
              Watcher {
                "active": true,
                "cb": [Function],
                "deep": false,
                "depIds": Set {},
                "deps": Array [],
                "dirty": false,
                "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                "getter": [Function],
                "id": 2,
                "lazy": false,
                "newDepIds": Set {},
                "newDeps": Array [],
                "sync": false,
                "user": false,
                "value": undefined,
                "vm": [Circular],
              },
            ],
          },
          "_self": [Circular],
          "_uid": 1,
          "_vnode": [Circular],
          "_watcher": Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {},
            "deps": Array [],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 2,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": false,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
          "_watchers": Array [
            Watcher {
              "active": true,
              "cb": [Function],
              "deep": false,
              "depIds": Set {},
              "deps": Array [],
              "dirty": false,
              "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
              "getter": [Function],
              "id": 2,
              "lazy": false,
              "newDepIds": Set {},
              "newDeps": Array [],
              "sync": false,
              "user": false,
              "value": undefined,
              "vm": [Circular],
            },
          ],
        },
        "data": undefined,
        "elm": <router-view />,
        "functionalContext": undefined,
        "functionalOptions": undefined,
        "functionalScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": false,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "router-view",
        "text": undefined,
      },
    ],
    "componentInstance": undefined,
    "componentOptions": undefined,
    "context": VueComponent {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        id="app"
      >
        <router-view />
      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "3": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_refElm": null,
        "components": Object {},
        "directives": Object {},
        "filters": Object {},
        "inject": Object {},
        "name": "app",
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_renderProxy": VueComponent {
        "$attrs": Object {},
        "$children": Array [],
        "$createElement": [Function],
        "$el": <div
          id="app"
        >
          <router-view />
        </div>,
        "$emit": [Function],
        "$listeners": Object {},
        "$options": Object {
          "_Ctor": Object {
            "3": [Function],
          },
          "_base": [Function],
          "_parentElm": null,
          "_refElm": null,
          "components": Object {},
          "directives": Object {},
          "filters": Object {},
          "inject": Object {},
          "name": "app",
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "$parent": undefined,
        "$refs": Object {},
        "$root": [Circular],
        "$scopedSlots": Object {},
        "$slots": Object {},
        "$vnode": undefined,
        "_c": [Function],
        "_data": Object {},
        "_directInactive": false,
        "_events": Object {},
        "_hasHookEvent": false,
        "_inactive": null,
        "_isBeingDestroyed": false,
        "_isDestroyed": false,
        "_isMounted": true,
        "_isVue": true,
        "_renderProxy": [Circular],
        "_self": [Circular],
        "_uid": 1,
        "_vnode": [Circular],
        "_watcher": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        "_watchers": Array [
          Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {},
            "deps": Array [],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 2,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": false,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
        ],
      },
      "_self": [Circular],
      "_uid": 1,
      "_vnode": [Circular],
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
    },
    "data": Object {
      "attrs": Object {
        "id": "app",
      },
    },
    "elm": <div
      id="app"
    >
      <router-view />
    </div>,
    "functionalContext": undefined,
    "functionalOptions": undefined,
    "functionalScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": false,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": "div",
    "text": undefined,
  },
}
`;
